plugins/modules/s3_object.py

```diff
@@ -1141,7 +1121,74 @@ def s3_object_do_copy(module, connection, s3_vars):
     if not s3_vars.acl_disabled:
         # only use valid object acls for the copy operation
         s3_vars.permission = s3_vars.object_acl
-    copy_object_to_bucket(module, connection, s3_vars.bucket, s3_vars.object, s3_vars.encrypt, s3_vars.metadata, s3_vars.validate, d_etag)
+    copy_object_to_bucket(
+        module, connection, s3_vars.bucket,
+        s3_vars.object, s3_vars.encrypt, s3_vars.metadata, s3_vars.validate, d_etag
+    )
+
+class VariableDict(dict):                    
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__dict__ = self
```

Comment:
We can avoid the constructor if `populate_facts()` returns a regular dict.

Reply:
I find it more convenient to access dict keys as dict.key instead of dict['key']. So I wrote this constructor.

Reply:
I think overriding the attribute space of a dictionary with the dictionary's data is something we should avoid. Imagine the following:
```python
d = VariableDict({"update": "foobar"})
d.update(some_other_dict)
```
If there's a desire to access a dictionary's items using dot notation there are better ways to do this, though I'd argue the convenience is not worth the effort.